quic/s2n-quic-core/src/recovery/bbr/probe_rtt.rs

```diff
@@ -90,56 +62,87 @@ impl BbrCongestionController {
-        self.bw_estimator.on_app_limited(bytes_in_flight);
+        bw_estimator.on_app_limited(bytes_in_flight);
 
-        if !self.probe_rtt_state.timer.is_armed() && bytes_in_flight <= self.probe_rtt_cwnd() {
+        if !self.timer.is_armed() && bytes_in_flight <= probe_rtt_cwnd {
             // Wait for at least ProbeRTTDuration to elapse:
-            self.probe_rtt_state.timer.set(now + PROBE_RTT_DURATION);
+            self.timer.set(now + PROBE_RTT_DURATION);
             // Wait for at least one round to elapse:
-            self.probe_rtt_state.round_done = false;
-            self.round_counter
-                .set_round_end(self.bw_estimator.delivered_bytes());
-        } else if self.probe_rtt_state.timer.is_armed() {
-            if self.round_counter.round_start() {
-                self.probe_rtt_state.round_done = true;
-            }
-            if self.probe_rtt_state.round_done {
-                self.check_probe_rtt_done(random_generator, now)
-            }
+            self.round_done = false;
+            round_counter.set_round_end(bw_estimator.delivered_bytes());
+        } else if self.timer.is_armed() && round_counter.round_start() {
+            self.round_done = true;
         }
     }
 
-    /// Checks if the ProbeRtt state should be exited and calls `exit_probe_rtt` if so
-    ///
-    /// The next ProbeRtt will be scheduled as necessary
-    pub fn check_probe_rtt_done<Rnd: random::Generator>(
+    /// Returns true if the `ProbeRtt` state is done and should be exited
+    pub fn is_done(&self, now: Timestamp) -> bool {
+        self.round_done && self.timer.is_expired(now)
+    }
+}
+
+/// Methods related to the `ProbeRtt` state
+impl BbrCongestionController {
+    /// Check if it is time to start probing for RTT changes, and enter the ProbeRtt state if so
+    pub fn check_probe_rtt<Rnd: random::Generator>(
         &mut self,
         random_generator: &mut Rnd,
         now: Timestamp,
     ) {
         //= https://tools.ietf.org/id/draft-cardwell-iccrg-bbr-congestion-control-02#4.3.4.4
-        //# BBRCheckProbeRTTDone()
-        //#    if (BBR.probe_rtt_done_stamp != 0 and
-        //#         Now() > BBR.probe_rtt_done_stamp)
-        //#       /* schedule next ProbeRTT: */
-        //#       BBR.probe_rtt_min_stamp = Now()
-        //#       BBRRestoreCwnd()
-        //#       BBRExitProbeRTT()
-
-        if self.probe_rtt_state.timer.poll_expiration(now).is_ready() {
-            /* schedule next ProbeRTT: */
-            self.data_volume_model.schedule_next_probe_rtt(now);
-            self.restore_cwnd();
-            self.exit_probe_rtt(random_generator, now);
+        //# BBRCheckProbeRTT()
+        //#    if (BBR.state != ProbeRTT and
+        //#         BBR.probe_rtt_expired and
+        //#         not BBR.idle_restart)
+        //#       BBREnterProbeRTT()
+        //#       BBRSaveCwnd()
+        //#       BBR.probe_rtt_done_stamp = 0
+        //#       BBR.ack_phase = ACKS_PROBE_STOPPING
```

Comment:
I don't see either of these being updated below

Reply:
`BBR.probe_rtt_done_stamp` is equivalent to `probe_rtt::State::timer`. Since `probe_rtt::State::timer` is part of the `probe_rtt::State` which only exists as part of the `bbr::State` enum, it does not exist when not in `State::ProbeRtt`. Therefore it does not need to be cancelled when entering `ProbeRtt`

`BBR.ack_phase` I have inside of the `probe_bw::State`. From the draft RFC, I only see the `ack_phase` being relevant while inside the `ProbeBw` state, so similar to `probe_rtt_done_stamp`, it only exists while in that state and thus doesn't need to be updated when entering ProbeRtt

I'll add some comments for this